oauth2-client-retrofit2/src/main/java/net/oauth2/client/retrofit/RetrofitTokenService.java

```diff
@@ -109,34 +107,31 @@ public RetrofitTokenService(final String serviceBaseUrl, final AccessTokenGrantR
 	}
 	
 	protected OkHttpClient httpClient(String userId, String password){
-		OkHttpClient httpClient = new OkHttpClient.Builder()
-				.addInterceptor(new Interceptor(){
-					@Override
-					public okhttp3.Response intercept(Chain chain) throws IOException {
-						Request request = chain.request();
-						String auth = request.header("Authorization");
-						if (auth == null) {
-							request = request.newBuilder()
-							.addHeader("Authorization", Credentials.basic(userId, password)).build();
-						}
-						return chain.proceed(request);
-					}})
+		return new OkHttpClient.Builder()
+				.addInterceptor(chain -> {
+					Request request = chain.request();
+					String auth = request.header("Authorization");
+					if (auth == null && !userId.equals("") && !password.equals("")) {
+						request = request.newBuilder()
+						.addHeader("Authorization", Credentials.basic(userId, password)).build();
+					}
```

Comment:
Okta does not want an authorization header.   When the values are not provided, don't create the authorization header.